### PR TITLE
chore(#8306): remove copy, sass, and ngtemplates

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Removing ngtemplates is the controversial bit. I couldn't find another library doing minification and bundling of angularjs templates and grunt was heavily used in the old library making forking difficult. I ended up rewriting it as a bespoke script.

medic/cht-core#8306

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8306-remove-grunt-5/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8306-remove-grunt-5/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8306-remove-grunt-5/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: scripts/build/build-angularjs-template-cache.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  for (const file of files) {
+    if (file === 'index.html') {
+      continue;
+    }
+    console.log(`Appending template: ${file}`);
+    const content = await getContent(file);
+    await writeBody(content);
+  }
+  await writeSuffix();
+})();
```
Comment (garethbowen) [inline scripts/build/build-angularjs-template-cache.js:53]: Custom replacement for ngtemplates. Not great, but easier than trying to fix the race condition!

File: admin/src/js/main.js [hunk 2]
```diff
@@ -124,7 +121,6 @@ require('./services/location');
 require('./services/modal');
 require('./services/moment-locale-data');
 require('./services/privacy-policies');
-require('./services/privacy-policies');
```
Comment (garethbowen) [inline admin/src/js/main.js:127]: Duplicated import for some reason.

File: scripts/build/build-angularjs-template-cache.js (hunk 1 above)
Comment (dianabarsan) [reply scripts/build/build-angularjs-template-cache.js:53]: I am having a very strong reaction to this :)) 
I agree, if this just works, there's not a lot of value in spending time to find some other solution (a librabry), which, for all intents and purposes, might be doing the same thing.

File: Gruntfile.js [hunk 3]
```diff
@@ -488,34 +398,33 @@ module.exports = function(grunt) {
...
 
   grunt.registerTask('copy-static-files-to-api', 'Copy build files and static files to api', [
-    'copy:api-bowser',
-    'copy:api-resources',
-    'copy:built-resources',
-    'copy:webapp-static',
-    'copy:admin-static',
+    'exec:copy-api-bowser',
+    'exec:copy-api-resources',
+    'exec:copy-webapp-static',
```
Comment (dianabarsan) [inline Gruntfile.js:407]: Do we no longer need to copy all built resources?

**Review by dianabarsan — COMMENTED** [review]
Just one question inline about one copy task I couldn't find.

File: Gruntfile.js (hunk 3 above)
Comment (garethbowen) [reply Gruntfile.js:407]: I... don't remember.

The command just moved everything from `build/static` to `api/build/static`. I've hunted around and it looks like everything now just compiles directly to `api/build/static` so it shouldn't be needed any more.

I haven't confirmed it 100% but I'd be shocked if it some static file was missed but the e2e tests still pass.

Comment (dianabarsan) [reply Gruntfile.js:407]: This is quite possible. Thanks for checking!

**Review by dianabarsan — APPROVED** [review]

**Pull request merged**